### PR TITLE
Tba/coding standards

### PR DESCRIPTION
**Pull request by timallen527**
- Setup environment to be able to run PHP_CodeSniffer.
- Added docker file for script execution.
- Updating readme for running PHP_CodeSniffer and formatter.
- Fixed formatting errors with auto formatting.
- Adding a github workflow action to run PHP_CodeSniffer.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -117,3 +117,24 @@ changed you can use the following command in that application:
...
+## Install Composer Dependencies
+
+```
+./bin/composer install
+```
+
+## Code Formatting Standards
+#### Run PHP_CodeSniffer to see the existing styling issues.
+```
+composer sniff
```
Comment (pcrist) [inline README.md]: Do these commands need the `./bin/` prefix?

File: composer.json [hunk 2]
```diff
@@ -13,12 +13,26 @@
             "role": "Developer"
         }
     ],
+    "repositories":[
```
Comment (pcrist) [inline composer.json]: Is this part necessary?

**Review by pcrist — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (timallen527) [reply README.md]: They do not as these script commands are defined in the `composer.json` file. 

That does raise a question of consistency though. We have some scripts that live in `bin` and can be ran from there. We have other scripts that are expected to be ran from the composer command. It'd be nice to settle on one way.

File: composer.json (hunk 2 above)
Comment (timallen527) [reply composer.json]: Currently, yes. The `coding-standards` package is not published to packagist.

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+services:
+  codesniffer:
+    image: php:8.2.4-cli-alpine
```
Comment (ha17) [inline docker-compose.yml]: `composer.json` requires `"php": ">=7.4.0"` so probably need to adapt this image to 7.4, too. Might need to alter the composer 2.4 image as well, to something compatible with php 7.4.

This probably will be a good image for it: [7.4-cli-alpine3.16](
https://hub.docker.com/layers/library/php/7.4-cli-alpine3.16/images/sha256-1e1b3bb4ee1bcb039f559adb9a3fae391c87205ba239b619cdc239b78b7f2557?context=explore)

Comment (ha17) [reply docker-compose.yml]: updated to a cli version ^^^

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    volumes:
+      - ./:/var/pdga
+    working_dir: /var/pdga
+
+  composer:
+    command: -V
+    image: composer:2.4
```
Comment (ha17) [inline docker-compose.yml:10]: Looks like this should work, compatibility for 2.4 is back to php 7.2 (https://github.com/composer/composer/tree/2.4.0)

File: docker-compose.yml (hunk 3 above)
Comment (ha17) [reply docker-compose.yml]: So, I'm wondering if I misread this. PHP >= is pretty broad. What is actually used in PRD? Is this package used in such a wide range that 7.4-8.2 is okay? I feel like CS will have different results based on the version of PHP, where 8.2 might have some stuff around nulls being passed to some functions that used to be allowed that aren't any more, and those might not get caught if we do change this. I guess I'm not sure the best way to handle that.

**Review by ha17 — APPROVED** [review]
I'm approving but the note about what PHP the codesniffer docker should have is something we might want to discuss.

File: README.md (hunk 1 above)
Comment (drew-acheson) [reply README.md]: We should neither expect nor require people to have composer installed locally.  I do not and do not wish to.  That's why I added the composer docker script to the PHP Codesniffer repo. I see that a composer docker script has been added already, so we should update the Readme to use that any time we run a composer command.

**Review by drew-acheson — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (timallen527) [reply README.md]: @drew-acheson This should be resolved in the latest update. Give me a shout if it gives you any issues.

File: composer.json (hunk 2 above)
Comment (timallen527) [reply composer.json]: The packagist package exists now and this repository specification has been removed.

File: docker-compose.yml (hunk 3 above)
Comment (timallen527) [reply docker-compose.yml:3]: I'm removing this as nothing is using the "codesniffer" service any longer.

File: README.md (hunk 1 above)
Comment (drew-acheson) [reply README.md]: Works great, thanks Tim!

**Review by drew-acheson — APPROVED** [review]

File: docker-compose.yml (hunk 3 above)
Comment (pcrist) [reply docker-compose.yml]: Prod in classic is on PHP 7.4.

php-api is on 8.2.4; that's defined in the repo in `Dockerfile.prod`.

The other repos typically have their PHP version defined in `docker-compose.yml` and it looks to be 8.1 or 8.2 across the board.

Comment (timallen527) [reply docker-compose.yml]: This has been updated to run using the  7.4-cli-alpine3.16 version mentioned above. 

The codesniffer commands have been updated to run specifically on this image.

**Comment by timallen527** [comment]
I chose to go away from using the composer scripts method of running the codesniffer commands. It is more consistent with our other commands and our other repos to just have the runnable scripts in `bin`.

**Review by ha17 — APPROVED** [review]

**Pull request merged**